executor/src/witgen/symbolic_evaluator.rs

```diff
@@ -15,4 +16,13 @@ impl<T: FieldElement> SymbolicVariables<T> for SymbolicEvaluator {
         // TODO arrays
         Ok(AffineExpression::from_variable_id(poly))
     }
+
+    fn challenge<'a>(
+        &self,
+        _challenge: &'a powdr_ast::analyzed::Challenge,
+    ) -> AffineResult<&'a AlgebraicReference, T> {
+        // TODO: Challenges can't be symbolically evaluated, because they can't be
+        // represented as an AffineExpression<&AlgebraicReference, T>...
+        Err(IncompleteCause::SymbolicExecutionOfChallenge)
+    }
```

Comment:
The base implementation of `SymbolicVariables::challenge` panics. For `FixedEvaluator` (which implements `SymbolicVariables`), this is still desired, because we should not encounter challenges when evaluating fixed columns. But `SymbolicVariables` is used to detect patterns in identities. So, for now, we cannot detect any patterns which include challenges, but at least it doesn't panic.